src/cart/crt_group.c

```diff
@@ -5076,6 +5078,7 @@ crt_group_secondary_modify(crt_group_t *grp, d_rank_list_t *sec_ranks,
 	d_rank_list_free(to_remove);
 	D_FREE(prim_idx);
 unlock:
+	grp_priv->gp_membs_ver = version;
```

Comment:
Perhaps we should only update gp_membs_ver if the whole modification operation succeeded? Here we may update gp_membs_ver when crt_group_mod_get fails, right? (Note that if no modification needs to be done, the opearation should be considered as successful and gp_membs_ver should be updated. DAOS may want to update the version ony.)

Reply:
the intention was to do only when operation succeeds, but didnt realize we jump to unlock in error condition - good catch, will change